File: backend/Projetos/KeySystems.ERP.ContasAPagar/src/KeySystems.ERP.ContasAPagar.Core/DomainModel/Fornecedor/Fornecedor.cs
namespace KeySystems.ERP.ContasAPagar.Core.DomainModel.Fornecedor
{
    public class Fornecedor : IEntity<int>
    {
        private Fornecedor() { }
        public Fornecedor(int id, string nomeBusca, string razaoSocial)
        {
            this.NomeBusca = nomeBusca;
            this.RazaoSocial = razaoSocial;
        }

        public int Id { get; set; }
        public string RazaoSocial { get; private set; }
        public string NomeBusca { get; private set; }
    }
}

File: frontend/src/app/modules/contas-pagar/fornecedor/shared/models/fornecedor.ts

export class Fornecedor 
{
    constructor(id : number, razaoSocial : string) {
        this.id = id;
        this.razaoSocial = razaoSocial;
    }

    private _id : number;
    public get id() : number {
        return this._id;
    }
    public set id(v : number) {
        this._id = v;
    }
    
    private _razaoSocial : string;
    public get razaoSocial() : string {
        return this._razaoSocial;
    }
    public set razaoSocial(v : string) {
        this._razaoSocial = v;
    }
}